src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -2210,6 +2199,40 @@ address StubGenerator::base64_AVX2_tables_addr() {
   return start;
 }
 
+address StubGenerator::base64_AVX2_decode_URL_tables_addr() {
```

Comment:
Shouldn't this be `decode_lut_tables`? As it's used for URL and non-URL decoding alike.

Reply:
These tables are used for both URL and non-URL based on the parameter, and they are only two of the three lut tables used (the other is in `base64_AVX2_decode_tables_addr` ).  Both names are essentially incorrect.  Does the name really matter that much?  It's the same as `base64_AVX2_decode_tables_addr` with the addition of URL tables.

Reply:
Names are important, but always hard to get right. At the very least they need to be correct. Maybe call it something like `..parameterized_decode_tables..` and the other `..shared_decode_tables..`?

Reply:
I prefer leaving them the way they are.  I don't think the names, along with the associated comments within the tables, causes any undue confusion as to their function.  However I will implement the name change if that's all it takes to procure a review approval.  Please provide the specific names you'd like me to use and I'll change them.  Or just approve as-is :-).

Reply:
I meant no disrespect here. By your own words `base64_AVX2_decode_URL_tables_addr` is "essentially incorrect", so I suggested some alternatives that I thought would make the code slightly more approachable. Regardless of whether you fix this detail I am not ready to approve this PR since I would need time to digest the latest changes. I'll ask a more senior engineer to review and give final approval (these changes need 2 reviewers approval anyhow).